src/main/java/hudson/tasks/junit/util/Polynomial.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Class:        Polynomial
+ * Description:
+ * Environment:  Java
+ * Software:     SSJ
+ * Copyright (C) 2001  Pierre L'Ecuyer and Universite de Montreal
+ * Organization: DIRO, Universite de Montreal
+ * @author       Éric Buist
+ * @since
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package hudson.tasks.junit.util;
+
+import java.io.Serializable;
+
+/**
+ * Represents a polynomial of degree n in power form. Such a polynomial
+ * is of the form
+ */
+public class Polynomial implements MathFunction, Serializable, Cloneable {
```

Comment:
can you annotate all of the annotated classes in this package with:
```suggestion
@Restricted(NoExternalUse.class)
public class Polynomial implements MathFunction, Serializable, Cloneable {
```

Which prevents other plugins from using them / prevents them from being in the public API of this plugin.

Reply:
Better just not to inline these classes I think. The primary motivation seems to have been to slim down the size of the JPI, but that can be done simply by excluding the transitive dependencies I think.

Reply:
Agreed, I did it and then started thinking. I didn't know you could exclude dependencies. Is this the way?

Reply:
Yes

Reply:
SSJ is now back as dependency.